File: bbarena-model/src/main/java/net/sf/bbarena/model/pitch/StreetBowlPitchFactory.java
package net.sf.bbarena.model.pitch;

import net.sf.bbarena.model.team.Team;

/**
 * Factory to create the Street Bowl Pitch
 *
 * @author f.bellentani
 */
public class StreetBowlPitchFactory extends PitchFactory {

	public static final String PITCH_NAME = "Street Bowl";
	public static final int WIDTH = 26;
	public static final int HEIGHT = 7;
	public static final int WIDE_ZONE = 2;

	public StreetBowlPitchFactory(Team... teams) {
		super(PITCH_NAME, WIDTH, HEIGHT, WIDE_ZONE, teams);
	}
}

File: src/main/kotlin/bbarena/model/rules/sb/StreetBowlPitchFactory.kt
package bbarena.model.rules.sb

import bbarena.model.rules.bb.PitchFactory
import bbarena.model.team.Team

val PITCH_NAME = "Street Bowl"
val WIDTH = 26
val HEIGHT = 7
val WIDE_ZONE = 2

fun buildPitch(teams: List<Team>) = PitchFactory(PITCH_NAME, WIDTH, HEIGHT, WIDE_ZONE, teams)
